File: redistributable/openstack.net/src/Samples/CPPCodeSamples/IdentityProviderExamples.cpp
#include "Stdafx.h"

using namespace net::openstack::Core::Domain;
using namespace net::openstack::Providers::Rackspace;
using namespace System;

ref class IdentityProviderExamples
{
public:
	void CreateProvider()
	{
#pragma region CreateProvider
		auto identity = gcnew CloudIdentity();
		identity->Username = "{username}";
		identity->APIKey = "{apiKey}";
		auto provider = gcnew CloudIdentityProvider(identity);
#pragma endregion
	}

	void CreateProviderWithPassword()
	{
#pragma region CreateProviderWithPassword
		auto identity = gcnew CloudIdentity();
		identity->Username = "{username}";
		identity->Password = "{password}";
		auto provider = gcnew CloudIdentityProvider(identity);
#pragma endregion
	}

	void CreateUser()
	{
		auto identity = gcnew CloudIdentity();
		identity->Username = "{username}";
		identity->APIKey = "{apiKey}";
		auto provider = gcnew CloudIdentityProvider(identity);

#pragma region CreateUser
		auto user = gcnew NewUser("{username}", "{email}", nullptr, true);
		user = provider->AddUser(user, nullptr);
		auto password = user->Password;
#pragma endregion
	}

	void UpdateUser()
	{
		auto identity = gcnew CloudIdentity();
		identity->Username = "{username}";
		identity->APIKey = "{apiKey}";
		auto provider = gcnew CloudIdentityProvider(identity);

#pragma region UpdateUser
		auto user = provider->GetUserByName("{username}", nullptr);
		user->Username = "{newUsername}";
		provider->UpdateUser(user, nullptr);
#pragma endregion
	}

	void ListUsers()
	{
		auto identity = gcnew CloudIdentity();
		identity->Username = "{username}";
		identity->APIKey = "{apiKey}";
		auto provider = gcnew CloudIdentityProvider(identity);

#pragma region ListUsers
		auto users = provider->ListUsers(nullptr);
		for each (User^ user in users)
			Console::WriteLine("{0}: {1}", user->Id, user->Username);
#pragma endregion
	}

	void AddRoleToUser()
	{
		auto identity = gcnew CloudIdentity();
		identity->Username = "{username}";
		identity->APIKey = "{apiKey}";
		auto provider = gcnew CloudIdentityProvider(identity);

#pragma region AddRoleToUser
		auto user = provider->GetUserByName("{username}", nullptr);
		provider->AddRoleToUser(user->Id, "{roleId}", nullptr);
#pragma endregion
	}

	void DeleteRoleFromUser()
	{
		auto identity = gcnew CloudIdentity();
		identity->Username = "{username}";
		identity->APIKey = "{apiKey}";
		auto provider = gcnew CloudIdentityProvider(identity);

#pragma region DeleteRoleFromUser
		auto user = provider->GetUserByName("{username}", nullptr);
		provider->DeleteRoleFromUser(user->Id, "{roleId}", nullptr);
#pragma endregion
	}

	void ResetApiKey()
	{
		auto identity = gcnew CloudIdentity();
		identity->Username = "{username}";
		identity->APIKey = "{apiKey}";
		auto provider = gcnew CloudIdentityProvider(identity);

#pragma region ResetApiKey
		auto credential = provider->ResetApiKey("{userId}", nullptr);
		auto newApiKey = credential->APIKey;
#pragma endregion
	}

	void ListRoles()
	{
		auto identity = gcnew CloudIdentity();
		identity->Username = "{username}";
		identity->APIKey = "{apiKey}";
		auto provider = gcnew CloudIdentityProvider(identity);

#pragma region ListRoles
		auto roles = provider->ListRoles(nullptr, Nullable<int>(), Nullable<int>(), nullptr);
		for each (Role^ role in roles)
			Console::WriteLine("{0}: {1}", role->Id, role->Name);
#pragma endregion
	}

	void DeleteUser()
	{
		auto identity = gcnew CloudIdentity();
		identity->Username = "{username}";
		identity->APIKey = "{apiKey}";
		auto provider = gcnew CloudIdentityProvider(identity);

#pragma region DeleteUser
		provider->DeleteUser("{userId}", nullptr);
#pragma endregion
	}
};

File: redistributable/openstack.net/src/Samples/VBCodeSamples/IdentityProviderExamples.vb
Imports net.openstack.Core.Domain
Imports net.openstack.Providers.Rackspace

Public Class IdentityProviderExamples

    Public Sub CreateProvider()
        ' #Region "CreateProvider"
        Dim identity = New CloudIdentity With {.Username = "{username}", .APIKey = "{apiKey}"}
        Dim provider = New CloudIdentityProvider(identity)
        ' #End Region
    End Sub

    Public Sub CreateProviderWithPassword()
        ' #Region "CreateProviderWithPassword"
        Dim identity = New CloudIdentity With {.Username = "{username}", .Password = "{password}"}
        Dim provider = New CloudIdentityProvider(identity)
        ' #End Region
    End Sub

    Public Sub CreateUser()
        Dim identity = New CloudIdentity
        Dim provider = New CloudIdentityProvider(identity)

        ' #Region "CreateUser"
        Dim User = New NewUser("{username}", "{email}", enabled:=True)
        User = provider.AddUser(User, Nothing)
        Dim password = User.Password
        ' #End Region
    End Sub

    Public Sub UpdateUser()
        Dim identity = New CloudIdentity
        Dim provider = New CloudIdentityProvider(identity)

        ' #Region "UpdateUser"
        Dim user = provider.GetUserByName("{username}", Nothing)
        user.Username = "{newUsername}"
        provider.UpdateUser(user, Nothing)
        ' #End Region
    End Sub

    Public Sub ListUsers()
        Dim identity = New CloudIdentity
        Dim provider = New CloudIdentityProvider(identity)

        ' #Region "ListUsers"
        Dim users = provider.ListUsers(Nothing)
        For Each user As User In users
            Console.WriteLine("{0}: {1}", user.Id, user.Username)
        Next
        ' #End Region
    End Sub

    Public Sub AddRoleToUser()
        Dim identity = New CloudIdentity
        Dim provider = New CloudIdentityProvider(identity)

        ' #Region "AddRoleToUser"
        Dim user = provider.GetUserByName("{username}", Nothing)
        provider.AddRoleToUser(user.Id, "{roleId}", Nothing)
        ' #End Region
    End Sub

    Public Sub DeleteRoleFromUser()
        Dim identity = New CloudIdentity
        Dim provider = New CloudIdentityProvider(identity)

        ' #Region "DeleteRoleFromUser"
        Dim user = provider.GetUserByName("{username}", Nothing)
        provider.DeleteRoleFromUser(user.Id, "{roleId}", Nothing)
        ' #End Region
    End Sub

    Public Sub ResetApiKey()
        Dim identity = New CloudIdentity
        Dim provider = New CloudIdentityProvider(identity)

        ' #Region "ResetApiKey"
        Dim credential = provider.ResetApiKey("{userId}")
        Dim newApiKey As String = credential.APIKey
        ' #End Region
    End Sub

    Public Sub ListRoles()
        Dim identity = New CloudIdentity
        Dim provider = New CloudIdentityProvider(identity)

        ' #Region "ListRoles"
        Dim roles = provider.ListRoles
        For Each role As Role In roles
            Console.WriteLine("{0}: {1}", role.Id, role.Name)
        Next
        ' #End Region
    End Sub

    Public Sub DeleteUser()
        Dim identity = New CloudIdentity
        Dim provider = New CloudIdentityProvider(identity)

        ' #Region "DeleteUser"
        provider.DeleteUser("{userId}", Nothing)
        ' #End Region
    End Sub

End Class
